File: 238. Product of Array Except Self.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        res = [1 for i in xrange(len(nums))]
        temp = 1
        for i in xrange(1, len(nums)):
            temp *= nums[i-1]
            res[i] *= temp
        temp = 1
        for i in xrange(len(nums)-1, -1, -1):
            if i<len(nums)-1:
                temp *= nums[i+1]
            res[i] *= temp
        return res

File: 238. Product of Array Except Self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector <int> res (nums.size(), 1);
        int temp = 1;
        for (int i = 1; i < nums.size(); i++)
        {
            temp *= nums[i-1];
            res[i] *= temp;
        }
        temp = 1;
        for (int i = nums.size()-2; i >= 0; i--)
        {
            temp *= nums[i+1];
            res[i] *= temp;
        }
        return res;
    }
};
